baybe/campaign.py

```diff
@@ -222,6 +243,68 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
+        self,
+        measurements: pd.DataFrame,
+        numerical_measurements_must_be_within_tolerance: bool,
+    ) -> None:
+        """Mark the given elements of the space as measured.
+
+        Args:
+            measurements: A dataframe containing parameter configurations to be
+                marked as measured.
+            numerical_measurements_must_be_within_tolerance: See
+                :func:`baybe.utils.dataframe.fuzzy_row_match`.
+        """
+        idxs_matched = fuzzy_row_match(
+            self.searchspace.discrete.exp_rep,
+            measurements,
+            self.parameters,
+            numerical_measurements_must_be_within_tolerance,
+        )
+        self._searchspace_metadata.loc[idxs_matched, _MEASURED] = True
+
+    def toggle_discrete_candidates(
+        self,
+        filter: pd.DataFrame,
+        exclude: bool,
+        anti: bool = False,
+        dry_run: bool = False,
+    ) -> pd.DataFrame:
+        """Un-/exclude certain discrete points from the candidate set.
+
+        Args:
+            filter: A dataframe specifying the filtering mechanism to determine the
+                candidates subset to be un-/excluded. The subset is determined via a
```

Comment:
what is uncluded?
```suggestion
                candidates subset to be in-/excluded. The subset is determined via a
```

Reply:
😄  this is not supposed to mean "uncluded" but "unexcluded". My thought was based on the way we currently write "un-/structuring" hooks everywhere, where the first version also means "unstructuring" and not "unructuring". But I see your point and probably we should be clear about what is the "correct" way to write stuff like this in english. Because one could also do stuff like (un-)structuring, etc. Do you know any resources / rules?

Reply:
I see, if this is the intent it is typically written like `(un-)excluded`

Reply:
OK, then I take away that I will also change `un-/structuring` hooks to `(un-)structuring hooks` next time I touch serialization.

But for this PR, probably we should then rather go with `in-/excluced` instead of `(un-)excluded`. Agree?

Reply:
Can also change the unstructuring part in this PR, if you prefer. It was only because it would touch a completely unrelated part of code

Reply:
Changed both things a two additional commits. Please thumbs up or resolve